### PR TITLE
implement FromValue on EnumValue

### PR DESCRIPTION
**Pull request by gdesmott**
not building:

error: lifetime may not live long enough
   --> glib/src/enums.rs:290:9
    |
285 | unsafe impl<'a> FromValue<'a> for &EnumValue {
    |             -- lifetime `'a` defined here
...
288 |     unsafe fn from_value(value: &'a Value) -> Self {
    |                                               ---- return type is &'1 EnumValue
289 |         let (_, v) = EnumValue::from_value(value).unwrap();
290 |         v
    |         ^ associated function was supposed to return data with lifetime `'1` but it is returning data with lifetime `'a`

### CODE REVIEW

File: glib/src/enums.rs [hunk 1]
```diff
@@ -282,6 +282,40 @@ impl Ord for EnumValue {
     }
 }
 
+unsafe impl<'a> FromValue<'a> for &EnumValue {
```
Comment (sdroege) [inline glib/src/enums.rs:285]:
```suggestion
unsafe impl<'a> FromValue<'a> for &'a EnumValue {
```

File: glib/src/enums.rs [hunk 2]
```diff
@@ -282,6 +282,43 @@ impl Ord for EnumValue {
     }
 }
 
+unsafe impl<'a> FromValue<'a> for &'a EnumValue {
```
Comment (sdroege) [inline glib/src/enums.rs:285]:
```suggestion
unsafe impl<'a, 'b> FromValue<'a> for &'b EnumValue {
```

**Review by sdroege — APPROVED** [review]

**Comment by gdesmott** [comment]
Added a test.

File: glib/src/enums.rs (hunk 2 above)
Comment (sdroege) [inline glib/src/enums.rs:285]:
```suggestion
unsafe impl<'a, 'b> FromValue<'a> for &'b EnumValue {
```

File: glib/src/enums.rs [hunk 3]
```diff
@@ -282,6 +282,43 @@ impl Ord for EnumValue {
...
+    type Checker = EnumTypeChecker;
+
+    unsafe fn from_value(value: &'a Value) -> Self {
+        let (_, v) = EnumValue::from_value(value).unwrap();
+        v
```
Comment (sdroege) [inline glib/src/enums.rs:290]:
```suggestion
        // SAFETY: The enum class and its values live forever
        unsafe { std::mem::transmute(v) }
```

**Review by sdroege — APPROVED** [review]

**Pull request merged**